immuni_analytics/apis/analytics.py

```diff
@@ -0,0 +1,278 @@
+
+
+@bp.route("/google/operational-info", methods=["POST"], version=1)
+@doc.summary("Upload android operational info (caller: Mobile Client.)")
+@doc.description(
+    "Check if the signed attestation is genuine and save the operational information in "
+    "the database."
+)
+@doc.consumes(doc.Boolean(name="Immuni-Dummy-Data", required=True), location="header")
+@doc.consumes(HeaderImmuniContentTypeJson(), location="header", required=True)
+@doc.consumes(GoogleOperationalInfo, location="body")
+@doc_exception(SchemaValidationException)
+@doc.response(
+    HTTPStatus.NO_CONTENT.value, None, description="Well-formed request.",
+)
+@validate(
+    location=Location.HEADERS,
+    is_dummy=IntegerBoolField(
+        required=True,
+        data_key="Immuni-Dummy-Data",
+        allow_strings=True,
+        description="Whether the current request is a dummy request. Dummy requests are ignored.",
+    ),
+)
+@validate(
+    location=Location.JSON,
+    province=Province(),
+    exposure_permission=IntegerBoolField(required=True),
+    bluetooth_active=IntegerBoolField(required=True),
+    notification_permission=IntegerBoolField(required=True),
+    exposure_notification=IntegerBoolField(required=True),
+    last_risky_exposure_on=IsoDate(),
+    salt=Base64String(required=True),
+    signed_attestation=fields.String(required=True),
+)
+async def post_android_operational_info(
+    request: Request,
+    is_dummy: bool,
+    province: str,
+    exposure_permission: bool,
+    bluetooth_active: bool,
+    notification_permission: bool,
+    exposure_notification: bool,
+    last_risky_exposure_on: date,
+    salt: str,
+    signed_attestation: str,
+) -> HTTPResponse:
+    """
+    Check if the signed attestation is valid and the salt has not been used
+    recently. In case of success save the operational_info.
+
+    :param request: the HTTP request object.
+    :param is_dummy: whether the uploaded data is dummy or not.
+    :param province: the user's province of residence.
+    :param exposure_permission: whether the user has granted the App permission to use the
+     Exposure Notifications framework.
+    :param bluetooth_active: whether the user has Bluetooth turned on.
+    :param notification_permission: whether the user has granted the App permission to display
+     notifications.
+    :param exposure_notification: whether the user was notified of a Risky Exposure after the
+     last Exposure Detection.
+    :param last_risky_exposure_on: the date on which the last Risky Exposure took place, if any.
+    :param salt: a random string sent by the client. Used to avoid replay attacks.
+    :param signed_attestation: the payload generated by Google SafetyNet attestation API.
+    :return: 200 in any case.
+    """
+    if is_dummy:
+        return json_response(body=None, status=HTTPStatus.NO_CONTENT)
+
+    if await managers.analytics_redis.get(safety_net.get_redis_key(salt)):
```

Comment:
Aren't you supposed to do a `setnx` as discussed yesterday?

Reply:
not here as it can still fail the validation

Reply:
Then we'd need a lock on that key, IMHO. 🤔